### PR TITLE
Add Enter and Tab back in to ios and macos key maps

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

I noticed that tab traversal stopped working on iOS when https://github.com/flutter/flutter/pull/73440 was landed.

It seems to be that the control characters were excluded from the list of logical keys, and so they ended up being generated values.

This PR just adds "Enter" and "Tab" to the list for both macOS and iOS.

## Tests

- Added a test for control characters to check their ID values to make sure they're using their ASCII values as the base for their ID and not something else.

### CODE REVIEW

**Comment by gspencergoog** [comment]
@dkwingsmt I noticed when I re-ran the script that I get some differences in the generated engine code too, including some that are somewhat unexpected: I assume I should be updating those too, right?

**Comment by dkwingsmt** [comment]
@gspencergoog I have some codegen changes collected in https://github.com/flutter/flutter/pull/81678. Let me submit that.

**Review by dkwingsmt — APPROVED** [review]
LGTM. I was reported this bug on macOS just now.

File: packages/flutter/test/services/keyboard_key_test.dart [hunk 1]
```diff
@@ -48,6 +49,16 @@ void main() {
       expect(LogicalKeyboardKey.isControlCharacter('~'), isFalse);
       expect(LogicalKeyboardKey.isControlCharacter('\xa0'), isFalse); // NO-BREAK SPACE
     });
+    test('Control characters are not using incorrect values', () async {
```
Comment (dkwingsmt) [inline packages/flutter/test/services/keyboard_key_test.dart:52]: Can you explain why you think this test is necessary?

**Pull request merged**